apps/wpcom-block-editor/src/wpcom/features/live-preview/upgrade-notice.tsx

```diff
@@ -0,0 +1,195 @@
+	);
+
+	const [ canPreviewButNeedUpgrade, setCanPreviewButNeedUpgrade ] = useState( false );
+	const [ themeType, setThemeType ] = useState< string >();
+
+	const siteEditorStore = useSelect( ( select ) => select( 'core/edit-site' ), [] );
+	const { createWarningNotice, removeNotice } = useDispatch( 'core/notices' );
+
+	const upgradePlan = useCallback( () => {
+		const generateCheckoutUrl = ( plan: string ) => {
+			return `${ getCalypsoUrl() }/checkout/${
+				window._currentSiteId
+			}/${ plan }?redirect_to=${ encodeURIComponent(
+				window.location.href
+			) }&checkoutBackUrl=${ encodeURIComponent( window.location.href ) }`;
+		};
+		const link =
+			themeType === WOOCOMMERCE_THEME
+				? generateCheckoutUrl( PLAN_BUSINESS ) // For a Woo Commerce theme, the users should have the Business plan or higher.
+				: generateCheckoutUrl( PLAN_PREMIUM ); // For a Premium theme, the users should have the Premium plan or higher.
+		window.location.href = link;
+
+		// TODO: Add the track event.
+	}, [ themeType ] );
+
+	/**
+	 * Get the theme and site info to decide whether the user needs to upgrade the plan.
+	 */
+	useEffect( () => {
+		if ( ! themeType ) {
+			wpcom.req
+				.get( `/themes/${ previewingThemeId }`, { apiVersion: '1.2' } )
+				.then( ( theme: Theme ) => {
+					const name = theme?.name;
+					if ( name ) {
+						setPreviewingThemeName( name );
+					}
+					return theme;
+				} )
+				.then( ( theme: Theme ) => {
+					const type = getThemeType( theme );
+					if ( ! type ) {
+						return;
+					}
+					setThemeType( type );
+				} )
+				.catch( () => {
+					// do nothing
+				} );
+			return;
+		}
+
+		/**
+		 * Currently, Live Preview only supports upgrades for Woo Commerce and Premium themes.
+		 */
+		if ( themeType !== WOOCOMMERCE_THEME && themeType !== PREMIUM_THEME ) {
+			setCanPreviewButNeedUpgrade( false );
+			return;
+		}
+
+		wpcom.req
+			.get( `/sites/${ window._currentSiteId }`, { apiVersion: '1.2' } )
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any
+			.then( ( site: any ) => {
+				if ( ! checkNeedUpgrade( { site, themeType } ) ) {
+					setCanPreviewButNeedUpgrade( false );
+					return;
+				}
+				setCanPreviewButNeedUpgrade( true );
+			} )
+			.catch( () => {
+				// do nothing
+			} );
+	}, [ previewingThemeId, themeType, setThemeType, setCanPreviewButNeedUpgrade ] );
+
+	useEffect( () => {
+		// Do nothing in the Post Editor context.
+		if ( ! siteEditorStore ) {
+			removeNotice( UPGRADE_NOTICE_ID );
+			return;
+		}
+
+		if ( ! isPreviewingTheme() ) {
+			removeNotice( UPGRADE_NOTICE_ID );
+			return;
+		}
+
+		if ( canPreviewButNeedUpgrade ) {
+			const type = themeType === WOOCOMMERCE_THEME ? 'Woo Commerce' : 'Premium';
+			createWarningNotice(
+				sprintf(
+					// translators: %s: The theme type ('Woo Commerce' or 'Premium')
+					__(
+						'You are previewing the %s theme that are only available after upgrading to the Premium plan or higher.',
```

Comment:
If this message is going to stay in the final version, I think that `that are` should be replaced with `that is` for it to be grammatically correct.